appsec/src/extension/tags.c

```diff
@@ -330,6 +340,14 @@ void dd_tags_rshutdown()
     }
 }
 
+bool should_collect_all_headers()
+{
+    return zend_llist_count(&_appsec_json_frags) > 0 ||
```

Comment:
`should_collect_all_headers()` is only called within an if statement which checks for `zend_llist_count(&_appsec_json_frags) == 0`, so this can never be true.

Reply:
Not now, but if there is an attack all headers should be collected

Reply:
But this function is not used for that purpose.

Reply:
It's not now but then `should_collect_all_headers` would be half true